### PR TITLE
emitBatchOverhead should only be used for splitting spans into batches

### LINKED ISSUE #2503: Jaeger exporter still creates too big UDP packets
### Description

This bug report is similar to a [previous bug report](https://github.com/open-telemetry/opentelemetry-go/issues/2465), this bug was fixed in [this PR](https://github.com/open-telemetry/opentelemetry-go/pull/2489) but the issue remains, potentially because of a small mistake in the PR.

The original bug was:
> Jaeger exporter logs errors like these: 'data does not fit within one UDP packet; size 65006, max 65000, spans 440'. This leads to spans missing from our traces.

But after the PR the bug still occurs and the logs now look as follows:

> 'data does not fit within one UDP packet; size 64941, max 64930, spans 435'

As you can see, the max went from 65000 to 64930. This is because of the new `emitBatchOverhead` of 70 introduced in the PR.
It seems that the `emitBatchOverhead` variable is not only used for limiting the amount of spans that get added (which is what was required to fix the original issue) but it is also used for the maximum size of an UDP packet (which is incorrect).
As seen in the error message max 64930 should be max 65000, which would fix the issue.

### Environment

- OS: alpine base image
- Architecture: x86
- Go Version: 1.17
- opentelemetry-go versions:
 go.opentelemetry.io/otel v1.3.1-0.20220104180153-776bfdcb2358
 go.opentelemetry.io/otel/exporters/jaeger v1.3.1-0.20220104180153-776bfdcb2358
 go.opentelemetry.io/otel/trace v1.3.1-0.20220104180153-776bfdcb2358

### Steps To Reproduce

1. Produce traces with a thousand spans.
2. Error 'data does not fit within one UDP packet; size 64941, max 64930, spans 435' is logged occasionally.

### Expected behavior

Jaeger exporter does not create UDP packets that are too big.

### PR DESCRIPTION
**Pull request by ken8203**
After apply this fix #2489, this kind of problem still occurs:

```
data does not fit within one UDP packet; size 64941, max 64930, spans 435
```

To solve this, `emitBatchOverhead` should only be used for splitting spans into batches.

resolve #2503

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Changed
 
 - Jaeger exporter takes into additional 70 bytes overhead into consideration when sending UDP packets (#2489)
+- Jaeger exporter takes into additional 70 bytes overhead into consideration only when splitting spans into batches (#2512)
```
Comment (Aneurysm9) [inline CHANGELOG.md:19]:
```suggestion
- Jaeger exporter takes into additional 70 bytes overhead into consideration when sending UDP packets (#2489, #2512)
```

I think there's really only need for one entry here.

File: exporters/jaeger/agent.go [hunk 2]
```diff
@@ -126,6 +126,11 @@ func (a *agentClientUDP) EmitBatch(ctx context.Context, batch *gen.Batch) error
 		// drop the batch if serialization of process fails.
 		return err
 	}
+
+	maxPacketSize := a.maxPacketSize
+	if maxPacketSize == udpPacketMaxLength {
+		maxPacketSize -= emitBatchOverhead
```
Comment (Aneurysm9) [inline exporters/jaeger/agent.go:132]:
```suggestion
	if maxPacketSize > udpPacketMaxLength - emitBatchOverhead {
		maxPacketSize = udpPacketMaxLength - emitBatchOverhead
```

If `maxPacketSize` were `udpPacketMaxLength - 1` there could be problems.

Comment (dashpole) [reply exporters/jaeger/agent.go:132]: I thought the same, but it also appears to currently be valid to set `maxPacketSize > udpPacketMaxLength` in your config.  If we also validate that `maxPacketSize <= udpPacketMaxLength`, then I think this makes sense.

File: CHANGELOG.md (hunk 1 above)
Comment (ken8203) [reply CHANGELOG.md]: good point 👍🏽

File: exporters/jaeger/agent.go (hunk 2 above)
Comment (ken8203) [reply exporters/jaeger/agent.go]: Yes, I agree with that. Adjusted!

**Review by dashpole — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Comment by ken8203** [comment]
Hi @dashpole and @Aneurysm9 

Could you please approve my running workflow since I'm a first-time contributor :D

**Review by MrAlias — APPROVED** [review]

**Pull request merged**